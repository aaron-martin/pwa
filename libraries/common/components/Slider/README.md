# Image Slider 
---

A slider component that allows putting multiple components into swipeable contexts.

> **Dependants:** <br> `<ImageSlider />`

## Getting Started

```
import { Slider } from 'Library/components';

<Slider>
  <Slider.Item>a</Slider.Item>
  <Slider.Item>b</Slider.Item>
  <Slider.Item>c</Slider.Item>
</Slider>
```

## Styling
An object with following keys can be passed as the classNames property.

### slider (optional)
Appends styling to the outer wrapper of the slider

### indicator (optional)
Appends styling to the outer wrapper of the 

### activeIndicator (optional)
Appends styling to the currently active dot

### inactiveIndicator (optional)
Appends styling to all other dots

## Props

### children (required)

_Type_: `React.Node`

`Slider.Item` that contain the content that is swipeable.

### disabled (optional)

_Type_: `bool`<br />
_Default_: `false`<br />

Whether swiping for this slider should be allowed.

###### Usage:

```
<Slider disabled>
  ...
</Slider>
```

### autoplay (optional) 

_Type_: `bool`<br>
_Default_: `false`<br>

If autoplay is enabled, the next slider item will appear automatically after `interval`ms

###### Usage:

```
<Slider autoplay>
  <Slider.Item>
    Hello
  </Slider.Item>
  <Slider.Item>
    World
  </Slider.Item>
</Slider>
```

### interval (optional) 

_Type_: `number`<br>
_Default_: `3000`<br>

The time specifies how many ms the autoplay needs to wait.
If autoplay is false, this prop is ignored.

###### Usage:

```
<Slider autoplay interval={2000}>
  <Slider.Item>
    Hello
  </Slider.Item>
  <Slider.Item>
    World
  </Slider.Item>
</Slider>
```

### controls (optional)

_Type_: `bool`<br>
_Default_: `false`<br>

Whether to display the control buttons or not

###### Usage:

```
<Slider controls>
  <Slider.Item>
    Hello
  </Slider.Item>
  <Slider.Item>
    World
  </Slider.Item>
</Slider>
```

### indicators (optional)

_Type_: `bool`<br>
_Default_: `false`<br>

Whether to display the indicators (little dots)

###### Usage:

```
<Slider indicators>
  <Slider.Item>
    Hello
  </Slider.Item>
  <Slider.Item>
    World
  </Slider.Item>
</Slider>
```

### loop (optional)

_Type_: `bool`<br>
_Default_: `false`<br>

If true then after the last item the first will appear again. 
And therefore allow swiping infinitely.

###### Usage:

```
<Slider loop>
  <Slider.Item>
    Hello
  </Slider.Item>
  <Slider.Item>
    World
  </Slider.Item>
</Slider>
```
