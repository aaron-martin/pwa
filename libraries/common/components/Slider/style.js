/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import { css } from 'glamor';

const sliderContainer = css({
  position: 'relative',
  maxHeight: '100%',
}).toString();

const sliderInnerContainer = css({
  overflow: 'hidden',
}).toString();

const slideWrapper = css({
  flexShrink: 0,
}).toString();

const sliderItem = css({
  position: 'relative',
  height: '100%',
}).toString();

export default {
  sliderContainer,
  sliderInnerContainer,
  slideWrapper,
  sliderItem,
};
