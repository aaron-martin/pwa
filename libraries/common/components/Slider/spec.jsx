/**
 * Copyright (c) 2017, Shopgate, Inc. All rights reserved.
 *
 * This source code is licensed under the Apache 2.0 license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { shallow, mount } from 'enzyme';
import Slider from './index';

jest.mock('swiper/dist/css/swiper.min.css', () => {});

describe('<Slider />', () => {
  const children = [
    <Slider.Item key="0"><div /></Slider.Item>,
    <Slider.Item key="1"><div /></Slider.Item>,
    <Slider.Item key="2"><div /></Slider.Item>,
  ];

  jest.useFakeTimers();

  it('renders with children', () => {
    const numChildren = children.length;
    const wrapper = shallow(<Slider>{children}</Slider>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(Slider.Item).length).toBe(numChildren);
  });

  it('renders with controls', () => {
    const wrapper = mount(<Slider controls>{children}</Slider>);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.swiper-button-next').length).toBe(1);
  });

  it('renders with indicators', () => {
    const wrapper = mount((
      <Slider indicators classNames={{ indicator: 'indicator-class' }}>{children}</Slider>
    ));

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('.indicator-class').length).toBe(1);
  });
});
