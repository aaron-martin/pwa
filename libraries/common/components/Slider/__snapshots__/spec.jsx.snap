// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> renders with children 1`] = `
<div
  className="css-1hlki0z "
>
  <ReactIdSwiper
    autoplay={null}
    containerClass="sg-swiper-container css-rdsogp "
    freeMode={false}
    initialSlide={0}
    loop={false}
    nextButton={null}
    onInit={[Function]}
    onSlideChangeEnd={[Function]}
    onTouchStart={[Function]}
    onTransitionEnd={[Function]}
    onTransitionStart={[Function]}
    pagination={null}
    paginationModifierClass="sg-swiper-pagination-"
    prevButton={null}
    slideClass="sg-swiper-slide"
    slidesPerView={1}
    wrapperClass="swiper-wrapper"
  >
    <div
      className="css-1l9a57c"
      key="s0"
    >
      <SliderItem
        key="0"
      >
        <div />
      </SliderItem>
    </div>
    <div
      className="css-1l9a57c"
      key="s1"
    >
      <SliderItem
        key="1"
      >
        <div />
      </SliderItem>
    </div>
    <div
      className="css-1l9a57c"
      key="s2"
    >
      <SliderItem
        key="2"
      >
        <div />
      </SliderItem>
    </div>
  </ReactIdSwiper>
</div>
`;

exports[`<Slider /> renders with controls 1`] = `
<Slider
  autoPlay={false}
  className=""
  classNames={Object {}}
  controls={true}
  disabled={false}
  indicators={false}
  initialSlide={0}
  interval={3000}
  loop={false}
  onSlideChange={null}
  slidesPerView={1}
  snapItems={true}
>
  <div
    className="css-1hlki0z "
  >
    <ReactIdSwiper
      autoplay={null}
      containerClass="sg-swiper-container css-rdsogp "
      freeMode={false}
      initialSlide={0}
      loop={false}
      nextButton=".swiper-button-next"
      onInit={[Function]}
      onSlideChangeEnd={[Function]}
      onTouchStart={[Function]}
      onTransitionEnd={[Function]}
      onTransitionStart={[Function]}
      pagination={null}
      paginationModifierClass="sg-swiper-pagination-"
      prevButton=".swiper-button-prev"
      slideClass="sg-swiper-slide"
      slidesPerView={1}
      wrapperClass="swiper-wrapper"
    >
      <div
        className="sg-swiper-container css-rdsogp "
      >
        <div
          className="swiper-wrapper"
        >
          <div
            className="sg-swiper-slide css-1l9a57c "
            key=".$s0"
          >
            <SliderItem
              key="0"
            >
              <div
                className="css-10a4qrv"
              >
                <div />
              </div>
            </SliderItem>
          </div>
          <div
            className="sg-swiper-slide css-1l9a57c "
            key=".$s1"
          >
            <SliderItem
              key="1"
            >
              <div
                className="css-10a4qrv"
              >
                <div />
              </div>
            </SliderItem>
          </div>
          <div
            className="sg-swiper-slide css-1l9a57c "
            key=".$s2"
          >
            <SliderItem
              key="2"
            >
              <div
                className="css-10a4qrv"
              >
                <div />
              </div>
            </SliderItem>
          </div>
        </div>
        <div
          className="swiper-button-next "
        />
        <div
          className="swiper-button-prev "
        />
      </div>
    </ReactIdSwiper>
  </div>
</Slider>
`;

exports[`<Slider /> renders with indicators 1`] = `
<Slider
  autoPlay={false}
  className=""
  classNames={
    Object {
      "indicator": "indicator-class",
    }
  }
  controls={false}
  disabled={false}
  indicators={true}
  initialSlide={0}
  interval={3000}
  loop={false}
  onSlideChange={null}
  slidesPerView={1}
  snapItems={true}
>
  <div
    className="css-1hlki0z "
  >
    <ReactIdSwiper
      autoplay={null}
      containerClass="sg-swiper-container css-rdsogp "
      freeMode={false}
      initialSlide={0}
      loop={false}
      nextButton={null}
      onInit={[Function]}
      onSlideChangeEnd={[Function]}
      onTouchStart={[Function]}
      onTransitionEnd={[Function]}
      onTransitionStart={[Function]}
      pagination=".indicator-class"
      paginationModifierClass="sg-swiper-pagination-"
      prevButton={null}
      slideClass="sg-swiper-slide"
      slidesPerView={1}
      wrapperClass="swiper-wrapper"
    >
      <div
        className="sg-swiper-container css-rdsogp "
      >
        <div
          className="swiper-wrapper"
        >
          <div
            className="sg-swiper-slide css-1l9a57c "
            key=".$s0"
          >
            <SliderItem
              key="0"
            >
              <div
                className="css-10a4qrv"
              >
                <div />
              </div>
            </SliderItem>
          </div>
          <div
            className="sg-swiper-slide css-1l9a57c "
            key=".$s1"
          >
            <SliderItem
              key="1"
            >
              <div
                className="css-10a4qrv"
              >
                <div />
              </div>
            </SliderItem>
          </div>
          <div
            className="sg-swiper-slide css-1l9a57c "
            key=".$s2"
          >
            <SliderItem
              key="2"
            >
              <div
                className="css-10a4qrv"
              >
                <div />
              </div>
            </SliderItem>
          </div>
        </div>
        <div
          className="indicator-class "
        />
      </div>
    </ReactIdSwiper>
  </div>
</Slider>
`;
